fdbkubernetesmonitor/kubernetes_test.go

```diff
@@ -0,0 +1,392 @@
+				enableNodeWatcher = true
+			})
+
+			It("should have the metadata for the pod and node", func() {
+				Expect(podClient.podMetadata).NotTo(BeNil())
+				Expect(podClient.nodeMetadata).NotTo(BeNil())
+				Expect(internalCache.InformersByGVK).To(HaveLen(2))
+			})
+		})
+	})
+
+	When("the PodClient handles events", func() {
+		BeforeEach(func() {
+			var err error
+			podClient, err = CreatePodClient(context.Background(), GinkgoLogr, enableNodeWatcher, func(fncNamespace string, fncPodName string, fncNodeName string) (client.WithWatch, cache.Cache, error) {
+				Expect(fncNamespace).To(Equal(namespace))
+				Expect(fncPodName).To(Equal(podName))
+				Expect(fncNodeName).To(Equal(nodeName))
+
+				return fakeClient, internalCache, nil
+			})
+
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("events for the pod are received", func() {
+			var fakeInformer *controllertest.FakeInformer
+			var pod *corev1.Pod
+
+			BeforeEach(func() {
+				pod = &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      podName,
+						Namespace: namespace,
+						Labels: map[string]string{
+							"testing": "testing",
+						},
+					},
+				}
+				var err error
+				fakeInformer, err = internalCache.FakeInformerFor(pod)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			When("an AddEvent is handled", func() {
+				BeforeEach(func() {
+					fakeInformer.Add(pod)
+				})
+
+				It("should update the pod information", func() {
+					Expect(podClient.podMetadata).NotTo(BeNil())
+					Expect(podClient.podMetadata.Name).To(Equal(podName))
+					Expect(podClient.podMetadata.Namespace).To(Equal(namespace))
+					Expect(podClient.podMetadata.Labels).To(Equal(map[string]string{
+						"testing": "testing",
+					}))
+				})
+			})
+
+			When("an UpdateEvent is handled", func() {
+				BeforeEach(func() {
+					fakeInformer.Update(nil, pod)
+				})
+
+				It("should update the pod information", func() {
+					Expect(podClient.podMetadata).NotTo(BeNil())
+					Expect(podClient.podMetadata.Name).To(Equal(podName))
+					Expect(podClient.podMetadata.Namespace).To(Equal(namespace))
+					Expect(podClient.podMetadata.Labels).To(Equal(map[string]string{
+						"testing": "testing",
+					}))
+
+					Expect(podClient.TimestampFeed).NotTo(Receive())
+				})
+			})
+
+			When("an UpdateEvent is handled that updates the OutdatedConfigMapAnnotation", func() {
+				var timestamp int64
+
+				BeforeEach(func() {
+					timestamp = time.Now().Unix()
+					pod.Annotations = map[string]string{
+						OutdatedConfigMapAnnotation: strconv.FormatInt(timestamp, 10),
+					}
+					fakeInformer.Update(nil, pod)
+				})
+
+				It("should update the pod information and receive an event", func() {
+					Expect(podClient.podMetadata).NotTo(BeNil())
+					Expect(podClient.podMetadata.Name).To(Equal(podName))
+					Expect(podClient.podMetadata.Namespace).To(Equal(namespace))
+					Expect(podClient.podMetadata.Labels).To(Equal(map[string]string{
+						"testing": "testing",
+					}))
```

Comment:
We already check the the timestamp feed as received the timestamp over the channel. Otherwise there is nothing to test as the `OnUpdate` event handler will not change the annotation.